models/enums_and_bitflags.py

```diff
@@ -0,0 +1,237 @@
+    output: str or None
+        Comma separated string with all the different ways the data is bad.
+        If given None, will return None.
+        If given zero, will return an empty string.
+    """
+    if value is None:
+        return None
+    elif value == 0:
+        return ''
+    elif isinstance(value, (int, float)):
+        output = []
+        for i in range(64):
+            if value >> i & 1:
+                if i not in dictionary:
+                    raise ValueError(f'Bitflag {i} not recognized in dictionary')
+                output.append(dictionary[i])
+        return ', '.join(output)
+    else:
+        raise ValueError(f'Bitflag must be integer/float, not {type(value)}')
+
+
+def string_to_bitflag(value, dictionary):
+    """
+    Takes a comma separated string, and converts it to a 64 bit integer bit-flag,
+    using the given dictionary (the inverse dictionary).
+    If any of the keywords are not recognized, will raise a ValueError.
+
+    To use this function, you must first define a dictionary with the keywords as keys,
+    and the corresponding bit-flag values as values. This should include all the possible keywords that could
+    be appended to this specific data model. For example, the badness bitflag for cutouts will not include
+    the keywords for images, as those will be set on the image model, not on the cutouts model.
+
+    If given an empty string, will return zero.
+    If given None, will return None.
+
+    Parameters
+    ----------
+    value: str or None
+        Comma separated string with all the different ways the data is bad.
+    dictionary: dict
+        Dictionary with the keywords as keys, and the corresponding bit-flag values as values.
+
+    Returns
+    -------
+    output: int or None
+        64 bit integer bit-flag.
+        If given None, will return None.
+        If given zero, will return an empty string.
+    """
+
+    if isinstance(value, str):
+        if value == '':
+            return 0
+        output = 0
+        for keyword in value.split(','):
+            original_keyword = keyword
+            keyword = keyword.lower().replace(' ', '')
+            if keyword not in dictionary:
+                raise ValueError(f'Keyword "{original_keyword}" not recognized in dictionary')
+            output += 2 ** dictionary[keyword]
+        return output
+
+
+image_badness_dict = {
+    1: 'Banding',
+    2: 'Shaking',
+    3: 'Saturation',
+    4: 'Bad Subtraction',
+    5: 'Bright Sky',
+}
+image_badness_inverse = {v.lower().replace(' ', ''): k for k, v in image_badness_dict.items()}
+
+cutouts_badness_dict = {
+    21: 'Cosmic Ray',
+    22: 'Ghost',
+    23: 'Satellite',
+    24: 'Offset',
+    25: 'Bad Pixel',
+    26: 'Bleed Trail',
+}
+cutouts_badness_inverse = {v.lower().replace(' ', ''): k for k, v in cutouts_badness_dict.items()}
+
+source_list_badness_dict = {
+    41: 'X-Match Failed',
+    42: 'Big Residuals',
+    43: 'Few Stars',
+    44: 'Many Stars',
```

Comment:
Too Many Stars?

Many stars doesn't sound like a bad thing to me.

Reply:
I agree to many sources is better. I think this sort of thing could happen when SExtractor fails and thinks every pixel is a source.

Reply:
I'd still recommend changing this to "too many sources" and "too few sources"

Reply:
Yes, I will do that. The beauty of this new setup is that we can make all the changes we want down the line. There's no need to commit to these enums at all. In fact we should probably discuss which values we want and how to name them.